src/unix/core.c

```diff
@@ -1739,25 +1739,40 @@ int uv__search_path(const char* prog, char* buf, size_t* buflen) {
   return UV_EINVAL;
 }
 
+#ifndef __linux__
+  uv__cpu_constraint uv__get_constrained_cpu(void) {
+    uv__cpu_constraint c = {0, 0, 0};
+    return c;
+  }
+#endif
+
 
 unsigned int uv_available_parallelism(void) {
 #ifdef __linux__
-  cpu_set_t set;
-  long rc;
+    cpu_set_t set;
+    long rc;
 
-  memset(&set, 0, sizeof(set));
+    memset(&set, 0, sizeof(set));
 
-  /* sysconf(_SC_NPROCESSORS_ONLN) in musl calls sched_getaffinity() but in
+    /* sysconf(_SC_NPROCESSORS_ONLN) in musl calls sched_getaffinity() but in
    * glibc it's... complicated... so for consistency try sched_getaffinity()
    * before falling back to sysconf(_SC_NPROCESSORS_ONLN).
-   */
-  if (0 == sched_getaffinity(0, sizeof(set), &set))
-    rc = CPU_COUNT(&set);
-  else
-    rc = sysconf(_SC_NPROCESSORS_ONLN);
-
-  if (rc < 1)
-    rc = 1;
+     */
+    if (0 == sched_getaffinity(0, sizeof(set), &set))
+        rc = CPU_COUNT(&set);
+    else
+        rc = sysconf(_SC_NPROCESSORS_ONLN);
+
+    uv__cpu_constraint c = uv__get_constrained_cpu();
+    if (c.period_length > 0) {
+        int rc_with_cgroup = (int)(c.quota_per_period /
+                                   c.period_length * c.proportions);
+        if (rc_with_cgroup < rc) {
+            rc = rc_with_cgroup;
+        }
```

Comment:
Style: var declarations go at the top; assignment separate from declaration; no braces around single if-statements

Substance: is the implicit cast to double intentional? `long long` can hold values >= 2**53, more than what double can accurately represent.